config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -1293,8 +1300,15 @@ public OpaqueTokenSpec introspectionUri(String introspectionUri) {
 			public OpaqueTokenSpec introspectionClientCredentials(String clientId, String clientSecret) {
 				Assert.hasText(clientId, "clientId cannot be empty");
 				Assert.notNull(clientSecret, "clientSecret cannot be null");
-				this.introspectionClientId = clientId;
-				this.introspectionClientSecret = clientSecret;
+				this.webClient = WebClient.builder()
+						.defaultHeader(HttpHeaders.AUTHORIZATION, basicHeaderValue(clientId, clientSecret))
```

Comment:
I think it would be easier to use this instead:

```
.defaultHeaders(h -> h.setBasicAuth(clientId, clientSecret))
```

You could then remove `basicHeaderValue`